docs/adr/0011-reporting-suspicious-activity.md

```diff
@@ -0,0 +1,98 @@
+This will allow the user to return to the activity page and see that the report is underway. It will prevent them from reporting multiple times.
+
+Once the initial state of the Report has been stored in DynamoDB the process to update all the back end systems can begin.
+
+The Report Suspicious Lambda will publish the original event to a SQS queue - Handle Suspicious Activity Queue. 
+
+The Handle Suspicious Activity Queue is subscribed by the Create Zendesk Ticket Lambda.  It will create a ticket in Zendesk using an HTTP call and then update DynamoDB with the ticket number.
+
+Once the ticket is created the lambda will publish a new event that includes to the original event and the ticket id.
+
+Multiple lambdas can listen for this event:
+
+1. Lambda to send an audit event to TxMA.  Once the event has been sent it will update DynamoDB with the TxMA event id.
+2. Lambda to optionally send an email via Notify.  Once the email has been sent, or rather requested to be sent, it will update DynamoDB with details from Notify.
+
+In failure scenarios:
+
+* The SQS visibility timeout on the queue will determine how long before the lambda can retry handling the event.
+* The maxReceiveCount determines how many times the lambda can try to handle the event before its sent to a DLQ.
+
+State machine would be modelled in DynamoDB.  We could use multiple fields in each activity record to show where we are in the process.  Having SQS separate the actions would mean we don't need to consult the state machine to determine what we can do next it would simply allow the front end to show where the process was up to.
+
+
+### Option 2 - Step Functions
+
+Define a Step Functions workflow that is activated by an event from SNS sent by the frontend and meeting the non-blocking requirement.  The Step Functions workflow will perform the required actions in order and in a resilient fashion
+
+#### Key features
+
+* Resilience is provided by Step Functions, each step must complete before the last and will be retried.  
+* Tasks can be run in parrallel, meeting the requirement for sending events and emails as concurrently.  
+* Additional steps can easily be specified.
+* Quick to define using a graphical workflow designer.
+* StepFunctions can be defined in SAM templates.
+* Resilience is a standard feature of Step Functions
+* Easy to see the progress of the workflow.
+* Built in debugging capability provides easier problem analysis.
+
+Cost may be higher than lambdas.  Example of 1000 workflows per month where each workflow has 10 state transitions would cost $0.15.
+
+### Option 3 - single lambda
+
+The whole process could be handled by a single lambda that picks up an event from SQS to decouple the front end.  The lambda would then perform each step required in turn:
+
+1. Update the database.
+2. Create the ticket.
+3. Store the ticket id in the database.
+4. Publish a TxMa event.
+5. Send an email via Notify.
```

Comment:
I'm not sure that this order is correct. I would have thought tickets 1, 4 and 5 could all be done in parallel (and have no dependencies on other tickets being done). Steps 2 and 3 are the only ones that have a dependency.

Reply:
I made updating the database the first task as we need to ensure a revisit to the activity page provides the correct information about the report.  Agree we could run tasks 1 and 2 in parallel.